storybook/CustomPropTable.js

```diff
@@ -0,0 +1,123 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import PropVal from '@storybook/addon-info/dist/components/PropVal'
+import PrettyPropType from '@storybook/addon-info/dist/components/types/PrettyPropType'
+
+const Table = props => <table {...props} />
+const Td = props => <td style={{paddingRight: 10, paddingTop: 10, textAlign: 'left', verticalAlign: 'top'}}
+  {...props} />
+const Tr = props => <tr {...props} />
+const Th = props => <th style={{paddingRight: 10, paddingTop: 6, textAlign: 'left', verticalAlign: 'top'}}
+  {...props} />
+const Tbody = props => <tbody {...props} />
+const Thead = props => <thead {...props} />
+
+export const multiLineText = input => {
+  if (!input) {
+    return input
+  }
+  const text = String(input)
+  const arrayOfText = text.split(/\r?\n|\r/g)
+  const isSingleLine = arrayOfText.length < 2
+  return isSingleLine
+    ? text
+    : arrayOfText.map((lineOfText, i) => (
+      // eslint-disable-next-line react/no-array-index-key
+      <span key={`${lineOfText}.${i}`}>
+        {i > 0 && <br />} {lineOfText}
+      </span>
+    ))
+}
+
+const determineIncludedPropTypes = (propDefinitions, excludedPropTypes) => {
+  if (excludedPropTypes.length === 0) {
+    return propDefinitions
+  }
+
+  return propDefinitions.filter(
+    propDefinition => !excludedPropTypes.includes(propDefinition.property)
+  )
+}
+
+export default function PropTable(props) {
+  const {
+    type,
+    maxPropObjectKeys,
+    maxPropArrayLength,
+    maxPropStringLength,
+    propDefinitions,
+    excludedPropTypes
+  } = props
+
+  if (!type) {
+    return null
+  }
+
+  const includedPropDefinitions = determineIncludedPropTypes(propDefinitions, excludedPropTypes)
+
+  if (!includedPropDefinitions.length) {
+    return <small>No propTypes defined!</small>
+  }
+
+  const propValProps = {
+    maxPropObjectKeys,
+    maxPropArrayLength,
+    maxPropStringLength
+  }
+
+  return (
+    <Table>
+      <Thead>
+        <Tr>
+          <Th>property</Th>
+          <Th>propType</Th>
+          <Th>required</Th>
+          <Th>default</Th>
+          <Th>description</Th>
+        </Tr>
+      </Thead>
+      <Tbody>
+        {includedPropDefinitions.map(row => (
+          <Tr key={row.property}>
+            <Td>{row.property}</Td>
+            <Td>
+              <PrettyPropType propType={row.propType} />
+            </Td>
+            <Td>{row.required ? 'yes' : '-'}</Td>
+            <Td>
+              {row.defaultValue === undefined ? (
+                '-'
+              ) : (
+                <PropVal val={row.defaultValue} {...propValProps} valueStyles={{}} />
+              )}
+            </Td>
+            <Td>{multiLineText(row.description)}</Td>
```

Comment:
multiLineText could be a react component

Reply:
I have chosen a straight forward implementation. I have changed the original code only if it was required for styling and not if it suitable.